include/sliding_window/search/search_setup.hpp

```diff
@@ -170,7 +169,13 @@ local_prefilter_fn::operator()(
                                                     window_size{arguments.window_size},
                                                     seed{adjust_seed(arguments.kmer_size)});
 
-    for (query_record const & record : records)
+    // if sequence smaller than pattern length it can't contain local match
+    auto pattern_length_filter = std::views::filter([&] (auto const & rec)
+    {
+        return rec.sequence.size() >= arguments.pattern_size;
+    });
+
+    for (query_record const & record : records | pattern_length_filter)
     {
         std::string const & id = record.sequence_id;
         std::vector<seqan3::dna4> const & seq = record.sequence;
```

Comment:
```suggestion
        std::vector<seqan3::dna4> const & seq = record.sequence;

        // if sequence smaller than pattern length it can't contain local match
        if (seq.size() < arguments.pattern_size)
            continue;
```

I would just make an additional "if" statement instead of a filter.